src/poetry/inspection/info.py

```diff
@@ -87,23 +86,17 @@ def __init__(
         self.requires_python = requires_python
         self.files = files or []
         self.yanked = yanked
-        self._cache_version = cache_version
         self._source_type: str | None = None
         self._source_url: str | None = None
         self._source_reference: str | None = None
 
-    @property
-    def cache_version(self) -> str | None:
-        return self._cache_version
-
     def update(self, other: PackageInfo) -> PackageInfo:
         self.name = other.name or self.name
         self.version = other.version or self.version
         self.summary = other.summary or self.summary
         self.requires_dist = other.requires_dist or self.requires_dist
         self.requires_python = other.requires_python or self.requires_python
         self.files = other.files or self.files
-        self._cache_version = other.cache_version or self._cache_version
         return self
 
     def asdict(self) -> dict[str, Any]:
```

Comment:
since we are now in Python 3.8 era, could you change this typing to use `TypedDict`? you could then eliminate `Any` in `CachedRepository._cache_func()`